### PR TITLE
Re-enable macOS .app export from Windows, add warnings about Unix permissions.

### LINKED ISSUE #96659: No longer able to export macOS builds as .app from Windows
### Tested versions

Godot 4.3

### System information

Windows 10 - Vulkan 1.3.242 - Forward+ - NVIDIA GeForce RTX 2060

### Issue description

In Godot 3 you could export mac builds as .app files, in Godot 4 that option is no longer there.

mac export g3
mac export g4

### Steps to reproduce

make any project and add a mac export option, then do Export Project

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/96659

### CODE REVIEW

File: platform/macos/export/export_plugin.cpp [hunk 1]
```diff
@@ -1903,6 +1899,11 @@ Error EditorExportPlatformMacOS::export_project(const Ref<EditorExportPreset> &p
 					if (is_executable(file)) {
 						// chmod with 0755 if the file is executable.
 						FileAccess::set_unix_permissions(file, 0755);
+#ifndef UNIX_ENABLED
+						if (export_format == "app") {
+							add_message(EXPORT_MESSAGE_INFO, TTR("Export"), vformat(TTR("Unable to set Unix permissions for executable \"%s\", use \"chmod +x\" to set it after transferring exported .app to macOS!"), "/Contents/MacOS/" + file.get_file()));
```
Comment (akien-mga) [inline platform/macos/export/export_plugin.cpp:1904]:
```suggestion
							add_message(EXPORT_MESSAGE_INFO, TTR("Export"), vformat(TTR("Unable to set Unix permissions for executable \"%s\". Use \"chmod +x\" to set it after transferring the exported .app to macOS or Linux."), "Contents/MacOS/" + file.get_file()));
```
Same for the other one.

**Comment by akien-mga** [comment]
Might be worth adding a note about this on https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_macos.html
The first paragraph describes the possible export formats, it could be expanded a bit to make clearer which ones can be exported from which platform, and what to do to make a `.app` exported from Windows actually usable on macOS.

**Comment by bruvzg** [comment]
> Might be worth adding a note about this

https://github.com/godotengine/godot-docs/pull/9921

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**